test/Generics/requirement_inference.swift

```diff
@@ -270,7 +270,7 @@ struct X18: P18, P17 {
 }
 
 // CHECK-LABEL: .X19.foo@
-// CHECK: Generic signature: <T, U where T == X18>
+// CHECK: Generic signature: <T, U where T == X18.A>
```

Comment:
This is just a change in printed sugar that doesn't affect the canonical signature. `X18.A` is `X18`, so we're ending up with this because of `T == T.A` vs `T == X18`.